docs/user/ppl/cmd/rex.rst

```diff
@@ -0,0 +1,199 @@
+=============
+rex
+=============
+
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Description
+============
+| The ``rex`` command extracts fields from a raw text field using regular expression named capture groups.
+
+Version
+=======
+3.3.0
+
+Syntax
+============
+rex [mode=<mode>] field=<field> <pattern> [max_match=<int>] [offset_field=<string>]
+
+* field: mandatory. The field must be a string field to extract data from.
+* pattern: mandatory string. The regular expression pattern with named capture groups used to extract new fields. Pattern must contain at least one named capture group using ``(?<name>pattern)`` syntax.
+* mode: optional. Either ``extract`` (default) or ``sed``. In extract mode, creates new fields from named capture groups. In sed mode, performs text substitution on the field.
+
+  **Sed mode syntax:**
+  
+  - **Substitution:** ``s/pattern/replacement/[flags]``
+    
+    - ``s/pattern/replacement/`` - Replace first occurrence
+    - ``s/pattern/replacement/g`` - Replace all occurrences (global)
+    - ``s/pattern/replacement/n`` - Replace only the nth occurrence (where n is a number)
+    - Backreferences: Use ``\1``, ``\2``, etc. to reference captured groups in replacement
+    
+  - **Transliteration:** ``y/from_chars/to_chars/`` - Character-by-character mapping
+
+* max_match: optional integer (default=1). Maximum number of matches to extract. If greater than 1, extracted fields become arrays.
+* offset_field: optional string. Field name to store the character offset positions of matches.
+
+Example 1: Basic Field Extraction
+==================================
+
+Extract username and domain from email addresses using named capture groups.
+
+PPL query::
+
+    os> source=accounts | rex field=email "(?<username>[^@]+)@(?<domain>[^.]+)" | fields email, username, domain | head 3 ;
+    fetched rows / total rows = 3/3
+    +--------------------------+--------------+---------+
+    | email                    | username     | domain  |
+    |--------------------------+--------------+---------|
+    | amberduke@pyrami.com     | amberduke    | pyrami  |
+    | hattiebond@netagy.com    | hattiebond   | netagy  |
+    | nanettebates@quility.com | nanettebates | quility |
+    +--------------------------+--------------+---------+
+
+
+Example 2: Handling Non-matching Patterns
+==========================================
+
+The rex command returns all events, setting extracted fields to null for non-matching patterns.
```

Comment:
the following test results should demo the null values.

Reply:
make sense. Fixed